### PR TITLE
Refactoring ospd code into roles

### PR DESCRIPTION
**Pull request by tkammer**

### CODE REVIEW

File: settings/installer/ospd.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+
+installer:
+    type: ospd
+    short_type: ospd
+    user:
```
Comment (obaranov) [inline settings/installer/ospd.yml]: user, undercloud, defaults,  and firewall keys can be removed from this file. These files are generated by Infrared according to the spec arguments

File: settings/installer/ospd/network/isolation/template/ipv4.yml [hunk 2]
```diff
@@ -1,24 +1,6 @@
 ---
 content:
     parameter_defaults:
-        InternalApiNetCidr: 172.17.1.0/24
-        InternalApiAllocationPools: [{'start': '172.17.1.10', 'end': '172.17.1.200'}]
-        InternalApiNetworkVlanID: 101
```
Comment (obaranov) [inline settings/installer/ospd/network/isolation/template/ipv4.yml]: These vlan's ID's are used by the loadbalancer role.

File: playbooks/installer/ospd/main.yml [hunk 3]
```diff
@@ -1,4 +1,81 @@
...
+        shell: "mkdir -p {{ installer.overcloud.template_base }}"
+
+      - name: create the base overcloud deploy script
+        template:
+            src: "templates/overcloud_deploy.sh.j2"
+            dest: "~/overcloud_deploy.sh"
+            mode: 0755
+  roles:
+      - {role: installer/ospd/overcloud/storage, when: "installer.storage is defined"}
+      - {role: installer/ospd/overcloud/network, when: "installer.network is defined and installer.overcloud.network.isolation.enable == 'yes'"}
```
Comment (obaranov) [inline playbooks/installer/ospd/main.yml]: According to the spec, the network.isolation propperty will be storred at installer.network.isolation namespace. 

Here is spec snippet: 

```
 network-isolation:
                    type: YamlFile
                    help: The overcloud network isolation type
                    required: yes
network-isolation-template:
                    type: YamlFile
                    help: The overcloud network isolation template

```

So installer.overcloud.network.isolation.enable should be changed to installer.network.isolation.enable

File: roles/installer/ospd/images/defaults/main.yml [hunk 4]
```diff
@@ -7,4 +7,4 @@ tar_files:
         - "ironic-python-agent.tar"
         - "overcloud-full.tar"
 
-tar_images: "{{ tar_files[installer.product.version] }}"
+tar_images: "{{ installer.images.files | default(tar_files[installer.product.version]) }}"
```
Comment (obaranov) [inline roles/installer/ospd/images/defaults/main.yml]: installer.images.files is no longer present in settings

File: roles/installer/ospd/images/tasks/build.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+---
+- name: Check if there is a guest image is present
+  stat:
+      path: "/home/{{ installer.user.name }}/guest_image.qcow2"
+  register: stat_result
+
+- name: Download guest image for build if needed
+  get_url:
+      dest: "/home/{{ installer.user.name }}/guest_image.qcow2"
+      url: "{{ installer.images.guest_image }}"
```
Comment (obaranov) [inline roles/installer/ospd/images/tasks/build.yml]: Installler does not have these settings. All the images info should be moved to the branch.

File: roles/installer/ospd/overcloud/network/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+---
+- include: "{{ installer.overcloud.network.isolation.type }}.yml"
```
Comment (obaranov) [inline roles/installer/ospd/overcloud/network/tasks/main.yml]: incorrect reference. Should be installler.network.isolation.type

File: roles/installer/ospd/undercloud/tasks/setup.yml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+      port: 22
+      host: "{{ ansible_ssh_host }}"
+      search_regex: OpenSSH
+      delay: 10
+  delegate_to: "{% if 'virthost' in groups %}virthost{% else %}localhost{% endif %}"
+  when: update|changed
+
+- name: install the oscplugin/tripleO client
+  yum:
+      name: "{% if installer.product.version == 7 %}python-rdomanager-oscplugin{% else %}python-tripleoclient{% endif %}"
```
Comment (obaranov) [inline roles/installer/ospd/undercloud/tasks/setup.yml]: All the values form spec are strings, so here product version will be '7'. I'd suggest to change the to 
```if installer.product.version == '7'```

File: playbooks/installer/ospd/main.yml (hunk 3 above)
Comment (tkammer) [inline playbooks/installer/ospd/main.yml]: done

**Comment by tkammer** [comment]
retest openstack

**Comment by tkammer** [comment]
retest openstack

**Comment by bregman-arie** [comment]
Openstack gate failure is not related.

**Pull request merged**